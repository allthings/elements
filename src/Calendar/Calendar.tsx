import React, { FunctionComponent } from 'react'

import UnstyledCalendar, {
  CalendarProps,
} from 'react-calendar/dist/entry.nostyle'
import { ColorPalette } from '@allthings/colors'
import insertCSS from '../utils/insertCSS'
import { color, lightness } from 'kewler'

insertCSS(`
.react-calendar {
  max-width: 100%;
  background: white;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 13px;
}
.react-calendar,
.react-calendar *,
.react-calendar *:before,
.react-calendar *:after {
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}
.react-calendar button {
  margin: 0;
  border: 0;
  outline: none;
  transition: 250ms;
  font-size: inherit;
}

.react-calendar button:enabled:hover {
  cursor: pointer;
}
.react-calendar__navigation {
  height: 44px;
}
.react-calendar__navigation button {
  min-width: 44px;
  background: none;
  transition: 250ms;

}
.react-calendar__navigation button:enabled:hover,
.react-calendar__navigation button:enabled:focus {
  background-color: ${ColorPalette.lightGrey};
}
.react-calendar__navigation button[disabled] {
  color: ${ColorPalette.text.gray};
}
.react-calendar__month-view__weekdays {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
}
.react-calendar__month-view__weekdays__weekday {
  padding: .5em;
}

.react-calendar__month-view__weekdays__weekday abbr {
  text-decoration: none;
}

.react-calendar__month-view__weekNumbers {
  font-weight: bold;
}
.react-calendar__month-view__weekNumbers .react-calendar__tile {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: .75em;
  padding: calc(1em) calc(0.6666666666666666em);
}
.react-calendar__month-view__days__day--weekend {}
.react-calendar__month-view__days__day--neighboringMonth {
  color: ${ColorPalette.grey};
}
.react-calendar__year-view .react-calendar__tile,
.react-calendar__decade-view .react-calendar__tile,
.react-calendar__century-view .react-calendar__tile {
  padding: 2em .5em;
}
.react-calendar__tile {
  max-width: 100%;
  text-align: center;
  padding: .75em .5em;
  background: none;
}
.react-calendar__tile:disabled,
.react-calendar button.blocked {
  color: ${ColorPalette.grey} !important;
  background: repeating-linear-gradient(
    135deg,
    ${ColorPalette.white},
    ${ColorPalette.white} 5px,
    ${ColorPalette.whiteIntense} 5px,
    ${ColorPalette.whiteIntense} 10px
  );
}
.react-calendar__tile:enabled:hover,
.react-calendar__tile:enabled:focus {
  color: white;
  background-color: ${ColorPalette.blue};
}
.react-calendar__tile--hasActive {
  background: ${color(ColorPalette.blue, lightness(20))}
}
.react-calendar__tile--hasActive:enabled:hover,
.react-calendar__tile--hasActive:enabled:focus {
  background: ${color(ColorPalette.blue, lightness(25))}
}
.react-calendar__tile--active {
  color: white;
  background: ${ColorPalette.blue};
}
.react-calendar__tile--active:enabled:hover,
.react-calendar__tile--active:enabled:focus {
  background: ${color(ColorPalette.blue, lightness(-20))}
}
.react-calendar--selectRange .react-calendar__tile--hover {
  background: ${color(ColorPalette.blue, lightness(35))}
}
`)

const notBlocked = (_: Date) => false
const notDisabled = (_: { activeStartDate: Date }) => false
/**
 * The `Calendar` component is a thin wrapper around https://github.com/wojtekmaj/react-calendar.
 */
const Calendar: FunctionComponent<ICalendarProps> = ({
  isBlockedDay = notBlocked,
  tileDisabled = notDisabled,
  ...props
}) => (
  <UnstyledCalendar
    {...props}
    // Add functionality for `isBlockedDay` to disable and color dates differently
    tileDisabled={info => isBlockedDay(info.date) || tileDisabled(info)}
    tileClassName={info => (isBlockedDay(info.date) ? 'blocked' : null)}
  />
)

export interface ICalendarProps extends CalendarProps {
  /** Indicates if a day is blocked. Expects a function which is invoked with a `Date` object and should return a boolean **/
  isBlockedDay?: (date: Date) => boolean
  /** Proxied from `react-calendar` prop types **/
  tileDisabled?: (date: { activeStartDate: Date }) => boolean
}

export default Calendar
