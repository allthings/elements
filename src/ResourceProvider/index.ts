export { default } from './ResourceProvider'
export * from './ResourceProvider'
