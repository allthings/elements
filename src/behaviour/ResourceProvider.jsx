import React from 'react'
import PropTypes from 'prop-types'

export const RESOURCE_PATH = 'https://static.allthings.me'

/**
 * Elements uses a set of static resources like images or icons.
 * In order to benefit from caching across all apps, these resources are provided by a static asset CDN.
 *
 * Whenever you like to use Icons or Illustrations, you need to use the ResouceProvider to let the components know where they are.
 */
class ResourceProvider extends React.Component {
  getChildContext() {
    return { resourcePath: this.props.resourcePath }
  }

  static childContextTypes = {
    resourcePath: PropTypes.string,
  }

  static propTypes = {
    resourcePath: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
  }

  static defaultProps = {
    resourcePath: RESOURCE_PATH,
    children: null,
  }

  render() {
    return this.props.children
  }
}

export default ResourceProvider
