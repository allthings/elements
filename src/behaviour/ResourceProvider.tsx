import React from 'react'

export const RESOURCE_PATH = 'https://static.allthings.me'

/**
 * Elements uses a set of static resources like images or icons.
 * In order to benefit from caching across all apps, these resources are provided by a static asset CDN.
 *
 * Whenever you like to use Icons or Illustrations, you need to use the ResouceProvider to let the components know where they are.
 */
const ResouceProvider = React.createContext({
  resourcePath: RESOURCE_PATH,
})

export type ResourceType = {
  resourcePath?: string
}

export default ResouceProvider
