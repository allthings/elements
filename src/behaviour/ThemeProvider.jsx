import React from 'react'
import PropTypes from 'prop-types'
import { Broadcast, Subscriber } from 'react-broadcast'
import { ColorPalette } from '@allthings/colors'

export const THEME_CHANNEL = 'theme'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.primary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

/**
* All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
* The ThemeProvider allows you to define the default colors for most elements.
*
* **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
**/
export default class ThemeProvider extends React.PureComponent {
  static propTypes = {
    root: PropTypes.bool,
    theme: PropTypes.shape({
      primary: PropTypes.string,
      text: PropTypes.string,
      secondaryText: PropTypes.string,
      titleColor: PropTypes.string,
      contrast: PropTypes.string,
      warn: PropTypes.string,
      disabled: PropTypes.string,
      background: PropTypes.string,
      textOnBackground: PropTypes.string,
    }),
    children: PropTypes.node,
  }

  static defaultProps = {
    theme: defaultTheme,
  }

  static contextTypes = Subscriber.contextTypes

  render() {
    let root = true
    if (this.context.broadcasts && this.context.broadcasts[THEME_CHANNEL]) {
      root = false
    }
    const { theme, children } = this.props
    return root ? (
      <Broadcast
        channel={THEME_CHANNEL}
        value={{
          ...ThemeProvider.defaultProps.theme,
          ...theme,
        }}
      >
        {children}
      </Broadcast>
    ) : (
      <Subscriber channel={THEME_CHANNEL}>
        {previousTheme => (
          <Broadcast
            channel={THEME_CHANNEL}
            value={{
              ...ThemeProvider.defaultProps.theme,
              ...previousTheme,
              ...theme,
            }}
          >
            {children}
          </Broadcast>
        )}
      </Subscriber>
    )
  }
}

export const ThemeShape = ThemeProvider.propTypes.theme

export const withTheme = (mapThemeToProps, displayName) => WrappedComponent =>
  // eslint-disable-next-line
  class extends React.PureComponent {
    static displayName = displayName || WrappedComponent.displayName
    static component = WrappedComponent

    renderComponent = theme => {
      const props = !mapThemeToProps
        ? { theme }
        : mapThemeToProps(theme, this.props)

      return <WrappedComponent {...this.props} {...props} />
    }

    render() {
      return (
        <Subscriber channel={THEME_CHANNEL}>{this.renderComponent}</Subscriber>
      )
    }
  }
