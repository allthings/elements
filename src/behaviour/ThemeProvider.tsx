import React, { FunctionComponent, createContext } from 'react'
import { ColorPalette } from '@allthings/colors'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.secondary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

export interface ITheme {
  primary: string
  text: string
  secondaryText: string
  titleColor: string
  contrast: string
  warn: string
  disabled: string
  background: string
  textOnBackground: string
  [key: string]: string
}

const ThemeContext = createContext<ITheme>(defaultTheme)
export const ThemeConsumer = ThemeContext.Consumer

/**
 * All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
 * The ThemeProvider allows you to define the default colors for most elements.
 *
 * **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
 **/
const ThemeProvider: FunctionComponent<{ theme?: Partial<ITheme> }> = ({
  children,
  theme = {},
}) => (
  <ThemeConsumer>
    {contextTheme => (
      <ThemeContext.Provider value={{ ...contextTheme, ...(theme as ITheme) }}>
        {children}
      </ThemeContext.Provider>
    )}
  </ThemeConsumer>
)

export default ThemeProvider

type mapThemeToPropsType = (theme: ITheme, props: any) => any

function createThemeConsumer<T>(
  mapThemeToProps: mapThemeToPropsType,
  WrappedComponent: React.ComponentType<T>,
  props: Omit<T, 'theme'>
) {
  return (theme: ITheme) => {
    const extraProps = !mapThemeToProps
      ? { theme }
      : mapThemeToProps(theme, props)

    return <WrappedComponent {...props} {...extraProps} />
  }
}

export function withTheme(mapThemeToProps: mapThemeToPropsType) {
  return function<T extends { theme: ITheme }>(
    WrappedComponent: React.ComponentType<T>
  ): React.ComponentType<Omit<T, 'theme'>> {
    return function(props: Omit<T, 'theme'>) {
      return (
        <ThemeConsumer>
          {createThemeConsumer<T>(mapThemeToProps, WrappedComponent, props)}
        </ThemeConsumer>
      )
    }
  }
}
