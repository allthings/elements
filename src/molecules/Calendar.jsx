import React from 'react'
import PropTypes from 'prop-types'

import UnstyledCalendar from 'react-calendar/dist/entry.nostyle'
import { ColorPalette } from '@allthings/colors'
import insertCSS from '../utils/insertCSS'

insertCSS(`
.react-calendar {
  max-width: 100%;
  background: white;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 13px;
}
.react-calendar,
.react-calendar *,
.react-calendar *:before,
.react-calendar *:after {
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}
.react-calendar button {
  margin: 0;
  border: 0;
  outline: none;
  transition: 250ms;
  font-size: inherit;
}

.react-calendar button.blocked {
  color: ${ColorPalette.red};
}

.react-calendar button:enabled:hover {
  cursor: pointer;
}
.react-calendar__navigation {
  height: 44px;
}
.react-calendar__navigation button {
  min-width: 44px;
  background: none;
  transition: 250ms;

}
.react-calendar__navigation button:enabled:hover,
.react-calendar__navigation button:enabled:focus {
  background-color: ${ColorPalette.lightGrey};
}
.react-calendar__navigation button[disabled] {
  color: ${ColorPalette.text.gray};
}
.react-calendar__month-view__weekdays {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
}
.react-calendar__month-view__weekdays__weekday {
  padding: .5em;
}

.react-calendar__month-view__weekdays__weekday abbr {
  text-decoration: none;
}

.react-calendar__month-view__weekNumbers {
  font-weight: bold;
}
.react-calendar__month-view__weekNumbers .react-calendar__tile {
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: .75em;
  padding: calc(1em) calc(0.6666666666666666em);
}
.react-calendar__month-view__days__day--weekend {}
.react-calendar__month-view__days__day--neighboringMonth {
  color: ${ColorPalette.greyIntense};
}
.react-calendar__year-view .react-calendar__tile,
.react-calendar__decade-view .react-calendar__tile,
.react-calendar__century-view .react-calendar__tile {
  padding: 2em .5em;
}
.react-calendar__tile {
  max-width: 100%;
  text-align: center;
  padding: .75em .5em;
  background: none;
}
.react-calendar__tile:disabled {
  color: ${ColorPalette.text.gray};
}
.react-calendar__tile:enabled:hover,
.react-calendar__tile:enabled:focus {
  background-color: #e6e6e6;
}
.react-calendar__tile--hasActive {
  background: #76baff;
}
.react-calendar__tile--hasActive:enabled:hover,
.react-calendar__tile--hasActive:enabled:focus {
  background: #a9d4ff;
}
.react-calendar__tile--active {
  background: #006edc;
  color: white;
}
.react-calendar__tile--active:enabled:hover,
.react-calendar__tile--active:enabled:focus {
  background: #1087ff;
}
.react-calendar--selectRange .react-calendar__tile--hover {
  background-color: #e6e6e6;
}
`)
/**
 * The `Calendar` component is a thin wrapper around https://github.com/wojtekmaj/react-calendar.
 */
const Calendar = ({ isBlockedDay, tileDisabled, ...props }) => (
  <UnstyledCalendar
    {...props}
    // Add functionality for `isBlockedDay` to disable and color dates differently
    tileDisabled={info => isBlockedDay(info.date) || tileDisabled(info)}
    tileClassName={info => (isBlockedDay(info.date) ? 'blocked' : null)}
  />
)

Calendar.propTypes = {
  ...UnstyledCalendar.propTypes,
  /** Indicates if a day is blocked. Expects a function which is invoked with a `Date` object and should return a boolean **/
  isBlockedDay: PropTypes.func,
  /** Proxied from `react-calendar` prop types **/
  tileDisabled: PropTypes.func,
}

Calendar.defaultProps = {
  isBlockedDay: () => false,
  tileDisabled: () => false,
}

export default Calendar
