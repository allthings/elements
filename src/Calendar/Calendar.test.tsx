import React from 'react'
import { render } from '@testing-library/react'
import ResourceProvider from '../ResourceProvider'
import ThemeProvider from '../ThemeProvider'
import Calendar from './Calendar'

describe('Calendar', () => {
  it('should be able to block days', () => {
    const { container } = render(
      <ThemeProvider>
        <ResourceProvider>
          <Calendar
            isBlockedDay={date =>
              new Date().toDateString() === date.toDateString()
            }
          />
        </ResourceProvider>
      </ThemeProvider>,
    )

    expect(container.querySelector('.blocked')!.textContent).toBe(
      String(new Date().getDate()),
    )
  })

  it('should render calendar', () => {
    const { container } = render(
      <ThemeProvider>
        <ResourceProvider>
          <Calendar value={new Date(2020, 0, 1)} />
        </ResourceProvider>
      </ThemeProvider>,
    )

    expect(container).toMatchSnapshot()
  })
})
