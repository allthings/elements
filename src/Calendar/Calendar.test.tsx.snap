// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render calendar 1`] = `
<div>
  <div
    class="react-calendar"
  >
    <div
      class="react-calendar__navigation"
      style="display: flex;"
    >
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
        type="button"
      >
        «
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
        type="button"
      >
        ‹
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__label"
        style="flex-grow: 1;"
        type="button"
      >
        January 2020
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
        type="button"
      >
        ›
      </button>
      <button
        aria-label=""
        class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
        type="button"
      >
        »
      </button>
    </div>
    <div
      class="react-calendar__viewContainer"
    >
      <div
        class="react-calendar__month-view "
      >
        <div
          style="display: flex; align-items: flex-end;"
        >
          <div
            style="flex-grow: 1; width: 100%;"
          >
            <div
              class="react-calendar__month-view__weekdays"
              style="display: flex; flex-wrap: no-wrap;"
            >
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Monday"
                  title="Monday"
                >
                  Mon
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Tuesday"
                  title="Tuesday"
                >
                  Tue
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Wednesday"
                  title="Wednesday"
                >
                  Wed
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Thursday"
                  title="Thursday"
                >
                  Thu
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Friday"
                  title="Friday"
                >
                  Fri
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Saturday"
                  title="Saturday"
                >
                  Sat
                </abbr>
              </div>
              <div
                class="react-calendar__month-view__weekdays__weekday"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
              >
                <abbr
                  aria-label="Sunday"
                  title="Sunday"
                >
                  Sun
                </abbr>
              </div>
            </div>
            <div
              class="react-calendar__month-view__days"
              style="display: flex; flex-wrap: wrap;"
            >
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 30, 2019"
                >
                  30
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="December 31, 2019"
                >
                  31
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__tile--active react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 1, 2020"
                >
                  1
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 2, 2020"
                >
                  2
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 3, 2020"
                >
                  3
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 4, 2020"
                >
                  4
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 5, 2020"
                >
                  5
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 6, 2020"
                >
                  6
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 7, 2020"
                >
                  7
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 8, 2020"
                >
                  8
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 9, 2020"
                >
                  9
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 10, 2020"
                >
                  10
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 11, 2020"
                >
                  11
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 12, 2020"
                >
                  12
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 13, 2020"
                >
                  13
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 14, 2020"
                >
                  14
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 15, 2020"
                >
                  15
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 16, 2020"
                >
                  16
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 17, 2020"
                >
                  17
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 18, 2020"
                >
                  18
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 19, 2020"
                >
                  19
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 20, 2020"
                >
                  20
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 21, 2020"
                >
                  21
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 22, 2020"
                >
                  22
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 23, 2020"
                >
                  23
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 24, 2020"
                >
                  24
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 25, 2020"
                >
                  25
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 26, 2020"
                >
                  26
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 27, 2020"
                >
                  27
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 28, 2020"
                >
                  28
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 29, 2020"
                >
                  29
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 30, 2020"
                >
                  30
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="January 31, 2020"
                >
                  31
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 1, 2020"
                >
                  1
                </abbr>
              </button>
              <button
                class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                type="button"
              >
                <abbr
                  aria-label="February 2, 2020"
                >
                  2
                </abbr>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
